frontend/common/src/components/ClientProvider/ClientProvider.tsx

```diff
@@ -81,10 +86,75 @@ const willAuthError = ({ authState }: { authState: AuthState | null }) => {
   return false;
 };
 
+// this is what the validation error extension looks like from our GraphQL server
+type ValidationExtension = {
+  validation: { [key: string]: [string] };
+  category: string;
+};
+
+// type guard to ensure the error extension looks like our ValidationExtension type
+function isValidationExtension(
+  extension: GraphQLErrorExtensions,
+): extension is ValidationExtension {
+  return (
+    (extension as ValidationExtension) !== undefined &&
+    extension.category === "validation"
+  );
+}
+
+// Accepts a CombinedError object and finds the validation error messages
+const extractValidationErrorMessages = (combinedError: CombinedError) =>
+  combinedError.graphQLErrors
+    .filter((graphQLError) => isValidationExtension(graphQLError.extensions))
+    .flatMap((graphQLError) => {
+      const validationExtension =
+        graphQLError.extensions as ValidationExtension;
+      return Object.keys(validationExtension.validation).map(
+        // one key per validation rule like "user.sub" or "user.email"
+        (key) => validationExtension.validation[key],
+      );
+    })
+    // Each rule has an array for possibly multiple error messages
+    .flatMap((messageArrays) => messageArrays);
+
+// Accepts a list of error messages, localizes them, and returns a formatted ReactNode for toasting
+const buildValidationErrorMessageNode = (
+  errorMessages: Array<string>,
+  intl: IntlShape,
+): ReactNode => {
+  const localizedMessages = errorMessages.map((errorMessage) => {
+    const localizedMessageDescriptor = tryFindMessageDescriptor(errorMessage);
+    if (localizedMessageDescriptor) {
+      return intl.formatMessage(localizedMessageDescriptor);
+    }
+    return errorMessage;
+  });
+
+  // if more than 1, toast a list
+  if (localizedMessages.length > 1) {
+    return (
+      <ul>
+        {localizedMessages.map((message) => (
+          <li key={message}>{message}</li>
+        ))}
+      </ul>
+    );
+  }
+
+  // if just 1, toast by itself
+  if (localizedMessages.length === 1) {
+    return <span>localizedMessages[0]</span>;
```

Comment:
I think this is the cause
should be in `{}`

Reply:
Doh, how did I miss that?  Good catch!